oshi-core/src/main/java/oshi/util/ArchitectureCodename.java

```diff
@@ -0,0 +1,36 @@
+package oshi.util;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+    /**
+     *
+     * Microarchitecture Codename and Processor Model and Family Numbers,
+     * eg. Family 6 Model 94
+     *
+     */
+
+
+public enum ArchitectureCodename {
+
+    /**
+     * Intel
+     */
+
+    SkyLake("6 94");
+
+    private String familyModel;
+
+    ArchitectureCodename(String familyModel){
+        this.familyModel = familyModel;
```

Comment:
depending on how you implement this you may want to separate family and model to two variables, as that's what data you'll have available from OSHI. (If you use as a lookup key you'll still end up combining them, but you could do that internally.)